src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -85,6 +114,36 @@ public static Map<String, String> parseStringToStringMap(XContentParser parser)
         return map;
     }
 
+    /**
+     * Parses an XContent object representing the object of LLMSpec
+     * @param parser An XContent parser whose position is at the start of the map object to parse
+     * @return instance of {@link org.opensearch.ml.common.agent.LLMSpec}
+     * @throws IOException parsing error
+     */
+    public static LLMSpec parseLLM(XContentParser parser) throws IOException {
```

Comment:
I assume `LLMSpec` has its own parse method, could we us that directly?

Reply:
Discussed with @amitgalitz offline. The LLMSpec parser throws a null value for `parameters` field mostly because of some private variable. That's why had to pull the parse method here.

Reply:
can you add a TODO here to figure this out or highlight this in code commment

Reply:
Added